compiler/lib/ocaml_compiler.ml

```diff
@@ -35,9 +39,12 @@ let rec constant_of_const : _ -> Code.constant =
       let l = List.map ~f:(fun f -> float_of_string f) sl in
       Float_array (Array.of_list l)
   | ((Const_pointer i) [@if ocaml_version < (4, 12, 0)]) ->
-      Int (Int32.of_int_warning_on_overflow i)
```

Comment:
(Does wasm_of_ocaml support ocaml < 4.12 ? the opam file says yes, but the CI doesn't test it.)

Reply:
In principle, yes, I think so.

Reply:
Probably not, actually, since there have been some runtime changes since then.
It should not be difficult to support these older versions of the compiler, but I'm not sure it is worth it. I should just bump the version in the opam file.
It's hard to test with the CI since it installs a version of `ppx_expect` that depends on ocaml 4.14.

Reply:
Anyway, it does not cost much to do the right thing here.